src/main/python/feature-extractor/server.py

```diff
@@ -110,19 +152,21 @@ def serve(port, workers):
         while True:
             time.sleep(_ONE_DAY_IN_SECONDS)
     except KeyboardInterrupt:
+        pool.terminate()
```

Comment:
This won't be called for exceptions different from `KeyboardInterrupt`. Actually I don't know whether it is going to leak so resources. Maybe you could use [this](https://docs.python.org/3.7/library/atexit.html).

Reply:
All sub-processes are created using fork. Operation system is going to take care of cleaning up when the server exits. So it's not a big deal. Also, we run feature-extractor in the container and any exit would cause restart of the container which would clean-up everything. 

I don't think we need to improve termination here as long as we don't have any real issue with how it works right now.